pkg/install/stack/kubermatic/stack.go

```diff
@@ -0,0 +1,274 @@
+	"k8c.io/kubermatic/v2/pkg/install/util"
+	"k8c.io/kubermatic/v2/pkg/log"
+	"k8c.io/kubermatic/v2/pkg/util/yamled"
+
+	storagev1 "k8s.io/api/storage/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	NginxIngressControllerChartName   = "nginx-ingress-controller"
+	NginxIngressControllerReleaseName = NginxIngressControllerChartName
+	NginxIngressControllerNamespace   = NginxIngressControllerChartName
+
+	CertManagerChartName   = "cert-manager"
+	CertManagerReleaseName = CertManagerChartName
+	CertManagerNamespace   = CertManagerChartName
+
+	DexChartName   = "oauth"
+	DexReleaseName = DexChartName
+	DexNamespace   = DexChartName
+
+	KubermaticOperatorChartName   = "kubermatic-operator"
+	KubermaticOperatorReleaseName = KubermaticOperatorChartName
+	KubermaticOperatorNamespace   = "kubermatic"
+
+	StorageClassName = "kubermatic-fast"
+)
+
+type Options struct {
+	HelmValues                 *yamled.Document
+	KubermaticConfiguration    *operatorv1alpha1.KubermaticConfiguration
+	RawKubermaticConfiguration *unstructured.Unstructured
+	ForceHelmReleaseUpgrade    bool
+}
+
+func Deploy(ctx context.Context, logger *logrus.Entry, kubeClient ctrlruntimeclient.Client, helmClient helm.Client, opt Options) error {
+	if err := deployStorageClass(ctx, logger, kubeClient); err != nil {
+		return fmt.Errorf("failed to deploy StorageClass: %v", err)
+	}
+
+	if err := deployNginxIngressController(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy nginx-ingress-controller: %v", err)
+	}
+
+	if err := deployCertManager(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy cert-manager: %v", err)
+	}
+
+	if err := deployDex(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy Dex: %v", err)
+	}
+
+	if err := deployKubermaticOperator(ctx, logger, kubeClient, helmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy Kubermatic Operator: %v", err)
+	}
+
+	if err := applyKubermaticConfiguration(ctx, logger, kubeClient, opt); err != nil {
+		return fmt.Errorf("failed to apply Kubermatic Configuration: %v", err)
+	}
+
+	return nil
+}
+
+func deployStorageClass(ctx context.Context, logger *logrus.Entry, kubeClient ctrlruntimeclient.Client) error {
+	logger.Infof("💾 Deploying %s StorageClass…", StorageClassName)
+
+	cls := storagev1.StorageClass{}
+	key := types.NamespacedName{Name: StorageClassName}
+
+	err := kubeClient.Get(ctx, key, &cls)
+
+	// storage class exists already
+	if err == nil {
+		logger.Info("✅ StorageClass exists, nothing to do.")
+		return nil
+	}
+
+	if !kerrors.IsNotFound(err) {
+		return fmt.Errorf("failed to check for StorageClass %s: %v", StorageClassName, err)
+	}
+
+	sc := storageClassForProvider(StorageClassName, "gke")
```

Comment:
Right now, gke is hardcoded which would need to be changed for a production version.
Tested with eks in my dev env, works fine.